### DISCUSSION TITLE
OpenCASCADE version Compatibility

### DISCUSSION THREAD

**Discussion by IbrahimSIM** [IfcOpenShell/IfcOpenShell#3379] [Q&A]
Hi everyone,

Is the OpenCASCADE 7.7.0 new version compatible with the latest IFC version (C++ SDK); the OpenCASCADE 7.5.0 compatible with the latest IFC version and there is no issues, but when use OpenCASCADE 7.7.0 with the latest version of IFC there are a random crashes occurring in the IfcParse.cpp.

**Comment by aothms** [comment]
> a random crashes occurring in the IfcParse.cpp.

IfcParse does not depend on OCCT, so there is something more intricate happening here. Can you provide some more details (platform, compilation method, full stack traces)

**Reply by IbrahimSIM** [reply]
The message that appear before the crash occur: 
"HEAP CORRUPTION Detected: after Normal block (#6069085) at 0x000002CCE65440.
CRT detected that the application wrote to memory after end of heap buffer."

**Reply by IbrahimSIM** [reply]
The platform: Win-10.
The Compilation Method: Debug x64 (Microsoft Visual Studio 2017).

**Reply by IbrahimSIM** [reply]
The crash often occur in delete_scalar.cpp
![Screenshot 2023-07-05 124416](https://github.com/IfcOpenShell/IfcOpenShell/assets/138665166/a2784f88-38c5-4458-bbf6-fc23c77d2690)

**Reply by IbrahimSIM** [reply]
This is the call stack:
![Screenshot 2023-07-05 124537](https://github.com/IfcOpenShell/IfcOpenShell/assets/138665166/e421a826-0011-4814-94a0-b02dac5aae52)

**Reply by IbrahimSIM** [reply]
Another call stack:
![2](https://github.com/IfcOpenShell/IfcOpenShell/assets/138665166/54e343ed-5bc1-4421-8839-30dea82f6638)

**Reply by IbrahimSIM** [reply]
Another call stack:
![3](https://github.com/IfcOpenShell/IfcOpenShell/assets/138665166/296ff700-db41-4c90-8b40-db667fa4a19c)

**Reply by IbrahimSIM** [reply]
Another call stack in xlocale
![6](https://github.com/IfcOpenShell/IfcOpenShell/assets/138665166/830fd8e7-387e-4b15-a0a5-66b2d9516ca3)
![4](https://github.com/IfcOpenShell/IfcOpenShell/assets/138665166/9703d1b2-ea2a-4d05-8014-4c7b166b7cde)

**Reply by aothms** [reply]
I'm guessing this is some stdlib incompatibility. Did you compile open cascade yourself using the same compiler toolset as your main application?

**Reply by IbrahimSIM** [reply]
We will rebuild all deps using VS2022 and see if the crashes are still there or not.
